### PR TITLE
Added Default logic to `files` cmd

### PR DESCRIPTION
**Pull request by RichardLitt**
Part of #2484

License: MIT
Signed-off-by: Richard Littauer <richard.littauer@gmail.com>

### CODE REVIEW

**Comment by Kubuxu** [comment]
There is an error in this PR:
http://ci.ipfs.team:8111/viewLog.html?buildId=1463&buildTypeId=GoIpfs_CiTests&tab=buildLog#_focus=2892

**Comment by RichardLitt** [comment]
That doesn't look like it is my fault.

File: core/commands/files/files.go [hunk 1]
```diff
@@ -417,11 +414,11 @@ Examples:
 			res.SetError(err, cmds.ErrNormal)
 			return
 		}
-		if found {
```
Comment (Kubuxu) [inline core/commands/files/files.go]: You removed this `if` block without removing `found` variable in line 412.

Comment (RichardLitt) [inline core/commands/files/files.go]: Fixed.

**Comment by Kubuxu** [comment]
Can we rerun tests on this?

**Comment by Kubuxu** [comment]
Here are still commands that miss some of the help texts:
http://ci.ipfs.team:8111/viewLog.html?buildId=1927&buildTypeId=GoIpfs_CiTests&tab=buildLog#_focus=2714

**Comment by RichardLitt** [comment]
I need some help with this; I am confused how you process this information. 

I open up Teamcity. I see a very long file. I see lots of yellow // orange text. Sometimes it is OK, but I don't know how I am supposed to know it is OK or not. I go to the bottom, and I see this:

```
[14:33:08][Step 4/4] make[1]: *** [t0250-files-api.sh] Error 1
[14:33:08][Step 4/4] 1..251
[14:33:08][Step 4/4] make: *** [test_sharness_expensive] Error 2
[14:33:08][Step 4/4] Makefile:29: recipe for target 't0250-files-api.sh' failed
[14:33:08][Step 4/4] make[1]: Leaving directory '/teamcity/work/3144e2a06f62fd48/src/github.com/ipfs/go-ipfs/test/sharness'
[14:33:08][Step 4/4] Makefile:98: recipe for target 'test_sharness_expensive' failed
[14:33:08][Step 4/4] Process exited with code 2
[14:33:10][Step 4/4] Step sharness (Command Line) failed
``

I don't know what code 2 is, or why the recipe failed. I go to t0250-files-api.sh in my editor. I don't know what to look for, I don't see anything that looks like a doc fix. 

What am I missing? How do you know what to look for in these tests outputs? What do you search for?

**Comment by Kubuxu** [comment]
You go to the bottom and look for red text: http://ci.ipfs.team:8111/viewLog.html?buildId=1966&buildTypeId=GoIpfs_CiTests&tab=buildLog#_state=22239&focus=22240

**Comment by RichardLitt** [comment]
@Kubuxu Yes; that's what I copied above. But that text isn't helpful for me, because I am missing some context (which is what I am trying to find out). I see that 0250 failed - but why?

**Comment by Kubuxu** [comment]
My link should go straight to a diff line: http://ci.ipfs.team:8111/viewLog.html?buildId=1966&buildTypeId=GoIpfs_CiTests&tab=buildLog&state=22239#_state=22239&focus=22237

0 bytes were expected, but command printed something.

**Comment by Kubuxu** [comment]
For some reason this fails here: http://ci.ipfs.team:8111/viewLog.html?buildId=2050&buildTypeId=GoIpfs_CiTests&tab=buildLog#_focus=22315

but I am not able to reproduce it:
```
 ~/git/go-ipfs   feature/add-default-to-files  cmd/ipfs/ipfs files read  /cat
Hello world
 ~/git/go-ipfs   feature/add-default-to-files  cmd/ipfs/ipfs files read  --count 0 /cat
 ~/git/go-ipfs   feature/add-default-to-files  
```

**Comment by whyrusleeping** [comment]
@RichardLitt unless you object strongly, i'm going to close this one its mostly adding `Default(false)` to boolean options, and we have agreed elsewhere that this isnt necessary

**Comment by RichardLitt** [comment]
@whyrusleeping There are a few things in this PR. Want me to split them out?

**Comment by whyrusleeping** [comment]
@RichardLitt yeah, that would be great

**Comment by RichardLitt** [comment]
Closing in favor of #3167 and #3168.

**Comment by asymmetric** [comment]
>  its mostly adding `Default(false)` to boolean options, and we have agreed elsewhere that this isnt necessary

Can I ask why that decision was made? I'm a first-time user of the IPFS daemon and having some flags explain the default value, and some not, is confusing to me.

**Comment by whyrusleeping** [comment]
@asymmetric the default value of any boolean option is false unless otherwise noted.

**Comment by asymmetric** [comment]
OK, but then shouldn't it be so consistently?

As you can see below, some options (e.g. `--init`) specify a `Default: false`, whereas others (e.g. `--migrate`) don't. That's what was confusing for me.

IPFS version **0.4.10**.

```
asymmetric@vps418113:~$ ipfs daemon --help
USAGE
  ipfs daemon - Run a network-connected IPFS node.

[...]

OPTIONS

  --init                         bool   - Initialize ipfs with default settings if not already initialized. Default: false.
  --mount                        bool   - Mounts IPFS to the filesystem. Default: false.
  --writable                     bool   - Enable writing objects (with POST, PUT and DELETE). Default: false.

  --unrestricted-api             bool   - Allow API access to unlisted hashes. Default: false.
  --disable-transport-encryption bool   - Disable transport encryption (for debugging protocols). Default: false.
  --enable-gc                    bool   - Enable automatic periodic repo garbage collection. Default: false.
  --manage-fdlimit               bool   - Check and raise file descriptor limits if needed. Default: true.
  --offline                      bool   - Run offline. Do not connect to the rest of the network but provide local API. Default: false.
  --migrate                      bool   - If true, assume yes at the migrate prompt. If false, assume no.
  --enable-pubsub-experiment     bool   - Instantiate the ipfs daemon with the experimental pubsub feature enabled.
  --enable-mplex-experiment      bool   - Add the experimental 'go-multiplex' stream muxer to libp2p on construction. Default: true.
```

**Comment by whyrusleeping** [comment]
@asymmetric Yeah, it should be consistent. Anywhere that says "default false" should get fixed

**Pull request closed without merging**